### PR TITLE
PERF: Checking monotonic-ness before sorting on an index #11080

### LINKED ISSUE #11080: PERF: checking is_monotonic_increasing/decreasing before sorting on an index
We don't keep the sortedness state in an index per-se, but it is rather cheap to check

- ``is_monotonic_increasing`` or ``is_monotonic_decreasing`` on a reg-index 
- MultiIndex should check ``is_lexsorted`` (this might be done already)

```
In [8]: df = DataFrame(np.random.randn(1000000,2),columns=list('AB'))

In [9]: %timeit df.sort_index()
10 loops, best of 3: 37.1 ms per loop

In [10]: %timeit -n 1 -r 1 df.index.is_monotonic_increasing
1 loops, best of 1: 2.01 ms per loop

In [11]: %timeit -n 1 -r 1 df.index.is_monotonic_increasin^C
KeyboardInterrupt

In [11]: %timeit df.set_index('A').sort_index()
10 loops, best of 3: 175 ms per loop

In [12]: %timeit -n 1 -r 1 df.set_index('A').index.is_monotonic_increasing
1 loops, best of 1: 9.54 ms per loop
```

### PR DESCRIPTION
**Pull request by scari**
closes #11080 

Worked for this during PyCon JP 2015 pandas sprint /w @sinhrks 

I found `test_frame.py:TestDataFrame.test_sort_index` cover this change.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3152,6 +3152,13 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
         else:
             from pandas.core.groupby import _nargsort
 
+            if (ascending and labels.is_monotonic_increasing) or \
```
Comment (jreback) [inline pandas/core/frame.py]: pls add a comment describing what you are doing here and a reference to the issue

**Comment by jreback** [comment]
pls add some benchmarks that I indicated in the issue (or point to ones that have this covered)

**Comment by scari** [comment]
Added a `frame_sort_index` test in `frame_methods.py`

-   30.96ms     4.64ms      0.15  frame_methods.frame_sort_index.time_frame_sort_index

**Comment by jorisvandenbossche** [comment]
@scari I think you did something wrong with rebasing (as there are a lot of other commits here). Can yo do

```
git fetch upstream
git rebase upstream/master
```

and then push again.

**Comment by scari** [comment]
@jorisvandenbossche  Thanks for your kind guidance! I just push again.

**Comment by jreback** [comment]
thanks @scari

**Pull request merged**